### PR TITLE
Optimizer switch to TBB

### PR DESCRIPTION
**Pull request by akenmorris**
This PR changes out OpenMP for TBB.  The main driver for this is to enable multithreading on Mac, which doesn't support OpenMP (at least not without great difficulty).  TBB has a nearly equivalent parallel-for construct.  I would prefer to use C++17's parallel-for, but we are compiling as C++14 for now (due to limited C++17 support on some platforms).  TBB may also allow us more control over threading than OpenMP should we choose to further parallelize the code.

* Note: This does not add an additional dependency as we already had TBB as a sub-dependency of OpenVDB.

* Diff of point files shows this to be identical before/after.  I did not run *all* the use cases, but I don't there is no reason to think that there would be a difference.

* Speed on windows/linux seems comparable (or faster).

* Speed on Mac is greatly improved.  For example, the left atrium use case runs 4x faster on my mac laptop.

* I left the USE_OPENMP option since there are still OpenMP pragmas in the trimesh code.

### CODE REVIEW

File: Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx [hunk 1]
```diff
@@ -109,24 +109,25 @@ namespace itk
...
           if (m_ParticleSystem->GetDomainFlag(dom) == true)
           {
-            continue;
+            return;
           }
 
           const ParticleDomain *domain = m_ParticleSystem->GetDomain(dom);
 
           typename GradientFunctionType::Pointer localGradientFunction = m_GradientFunction;
 #ifdef SW_USE_OPENMP
```
Comment (medakk) [inline Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx:127]: can this #ifdef be removed?

Comment (akenmorris) [reply Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx:127]: Yes indeed, and the others.

**Comment by akenmorris** [comment]
ellipsoid_mesh is running quite a bit faster with TBB and I'm not sure why.  The results _look_ fine, but the numbers are a bit different.  However, I am discovering that with OpenMP enabled, the mesh code is not stable in the master branch.  Investigating...

**Comment by akenmorris** [comment]
Ok, confirmed after dropping the random placement on the mesh, now results from single and multi-threaded are the same and reproducible.

This should be ready to merge.

**Comment by medakk** [comment]
Tested on ellipsoid, the resulting shape models have identical compactness, generalization and specificity.

**Review by medakk — APPROVED** [review]

**Pull request merged**